src/v/kafka/client/types.h

```diff
@@ -24,6 +24,23 @@ struct record_essence {
     std::optional<iobuf> key;
     std::optional<iobuf> value;
     std::vector<model::record_header> headers;
+
+    std::size_t size_bytes() const {
+        std::size_t sz = 0;
+        if (partition_id) {
+            sz += sizeof(model::partition_id);
+        }
```

Comment:
It's measuring its estimated size when serialized as an `iobuf`

Reply:
There's an API of the kafka client where raw data is passed in (`client::produce(std::vector<record_essence>`), later on the internal batch builder within the client moves the iobuf to an internal batch builder. So the size can't be known by the caller.

Reply:
what I mean is wouldn't the batch builder be the place to centralize things like estimates of batch size? Not that someone that wants that estimate needs to go through the trouble of building an entire builder.

Reply:
Yeah i agree, its just that even the batch builder doesn't actually call build until it is about to send a batch.

Are you saying that's fine and maybe this calculation should belong within the batch builder anyway?  I was considering that but I didn't want to edit the client unless it was necessary.

Reply:
leaving a comment on the interface is also acceptable. someone looks at that interface in the future then they should have the benefit of this conversation already having occurred.

Reply:
On second thought i'm thinking about dropping this commit , i'm not populating any of these fields instead of value, so i could just query `buf.value()` for a good enough estimate